lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -119,6 +119,8 @@
 
                     this.teamModel.save(data, { wait: true })
                         .done(function(result) {
+                            // So that the topics list can be refreshed.
+                            Backbone.trigger('teamModel:add', view.teamModel);
```

Comment:
We are seeing a bug where the "My Teams" page doesn't reflect your new membership after creating a team. For non-privileged users, creating a team will automatically add them as a member of the new team. I think that the simplest option might be to forcibly refresh the team membership collection for all users upon creation of a team so that the client doesn't have to know the rules.